wiki/bin/standupbot.py

```diff
@@ -0,0 +1,162 @@
+# StandupBot
+
+from __future__ import print_function
+from future.standard_library import install_aliases
+install_aliases()
+
+import re
+import json
+import random
+import datetime
+
+from urllib.parse import urlparse, urlencode
+from urllib.request import urlopen, Request
+from urllib.error import HTTPError
+
+from mediawiki import MediaWiki, CONFIG
+
+# Handy flags to help with debugging...
+# Don't actually push messages to Slack, just show the message payload locally
+DO_NOT_SEND = False
+# Add all standups to the announce list, regardless of time
+ANNOUNCE_ALL = False
+# Announce in a specific channel, rather than the one listed in the Wiki
+# False is a default value, but '@yourusername' and '#sandbox' are also good
+ANNOUNCE_IN = False
+
+print('StandupBot')
+
+# First check, let's see if we want to run at all
+
+if datetime.datetime.today().weekday() < 5:
+    print('Today is a weekday, starting the process!')
+
+    site = MediaWiki(user_agent='StandupBot')
+
+    # Regex patterns
+    listPattern = re.compile('\* (.*)')
+    sectionPattern = re.compile('<!-- Begin Standups Schedule -->(.*?)<!-- End Standups Schedule -->', re.M | re.S)
+    standupPattern = re.compile('\|-\n\| (.*?)\n\| (.*?)\n\| (.*?)\n\| (.*?)\n', re.M | re.S)
+
+    announceForTime = datetime.datetime.now() + datetime.timedelta(minutes=1)
```

Comment:
IMO this would be better if it announced standups at the time they start, as opposed to a minute early.